crates/state/src/lib.rs

```diff
@@ -0,0 +1,387 @@
+mod error;
+
+use sqlx::SqlitePool;
+
+use std::{collections::HashMap, convert::TryInto};
+
+use svm_hash::{Blake3Hasher, Hasher};
+
+pub use error::{GlobalStateError, Result};
+
+/// Every commit
+pub type Commit = [u8; 32];
+
+const SQL_SCHEMA: &str = include_str!("resources/schema.sql");
+
+fn hash_key_value_pair(key_hash: &[u8; 32], value: &[u8]) -> [u8; 32] {
+    let mut hasher = Blake3Hasher::default();
+    hasher.update(key_hash);
+    hasher.update(value);
+    hasher.finalize()
+}
+
+fn xor_signature(sig_1: &mut [u8; 32], sig_2: &[u8; 32]) {
+    for (a, b) in sig_1.iter_mut().zip(sig_2) {
+        *a ^= *b;
+    }
+}
+
+const ZERO_SIGNATURE: [u8; 32] = [0; 32];
+
+/// Some external resources:
+///
+/// * https://www.programmersought.com/article/33363860077/
+/// * https://www.youtube.com/watch?v=oEpY4NkkeYQ
+///
+/// Lists of namespaces within [`GlobalState`]:
+///
+/// * 't': templates
+///   * ''
+/// * 'a': accounts
+///   * 'b': balance
```

Comment:
Why do we need a namespace for `balance`, `nonce`?

Reply:
Just examples and fillers for now.